patches/net/minecraft/world/entity/Entity.java.patch

```diff
@@ -292,6 +292,19 @@
      }
  
      public boolean is(Entity p_20356_) {
+@@ -2451,10 +_,11 @@
+     }
+ 
+     public boolean isInvulnerableTo(DamageSource p_20122_) {
+-        return this.isRemoved()
++        boolean isVanillaInvulnerable = this.isRemoved()
+             || this.invulnerable && !p_20122_.is(DamageTypeTags.BYPASSES_INVULNERABILITY) && !p_20122_.isCreativePlayer()
+             || p_20122_.is(DamageTypeTags.IS_FIRE) && this.fireImmune()
+             || p_20122_.is(DamageTypeTags.IS_FALL) && this.getType().is(EntityTypeTags.FALL_DAMAGE_IMMUNE);
++        return net.neoforged.neoforge.common.CommonHooks.onEntityInvulnerablityCheck(this, p_20122_, isVanillaInvulnerable);
```

Comment:
The only overrides where this occurs is:
- Warden: checks if digging animation is active and damage source bypasses immunity
- Ghast: checks entity invulnerability, same damage source check, and if the damage is not a deflected fireball
- Breeze: damage source is in the breeze immune tag or source is from a breeze (self included)

I am of the opinion that the immunity bypass tags are compatible with this feature mainly because they are configurable.      A user or modder still has control over the tag contents to handle immunity if it matters for these 3 mobs.  The other aspects relate to reasonable invulnerabilities (breeze not affecting itself, warden (de)spawning).  The Ghast fireball is the only other one that might be worth patching.  I agree that patching each of these mobs for the sake of uniformity would be untenable in the long run though.

> I don't think we can reasonably provide the isVanillaInvulerable field since it is not necessarily correct
Perhaps rename?  This way it's clearer that it isn't absolute in its scope.